### PR TITLE
chore: enable labels for api calls

### PR DESCRIPTION
**Pull request by sanpj2292**
## What are the changes introduced in this PR?

Add below labels to any API calls made from transformer
- statusCode
- destinationId
- workspaceId

Metadata has been propagated for intermediate API calls(to obtain extra metadata in metrics) for below destinations have been added
- intercom
- mautic
- optimizely_fullstack
- active_campaign
- klaviyo
- delighted

## What is the related Linear task?

Resolves INT-2152

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3476/test-report.html

**Review by utsabc — DISMISSED** [review]

**Review by utsabc — DISMISSED** [review]

File: src/v0/util/index.js [hunk 1]
```diff
@@ -2258,6 +2258,14 @@ const applyCustomMappings = (message, mappings) => {
   );
 };
 
+const getRelativePathFromURL = (inputUrl) => {
```
Comment (koladilip) [inline src/v0/util/index.js:2261]: can you add a comment that this should only be used when URL path doesn't have any dynamic params like account ID etc

File: src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml [hunk 2]
```diff
@@ -59,7 +61,9 @@ steps:
     description: Fetch the data file from the data file url
     template: |
       const dataFileUrl = .destination.Config.dataFileUrl;
-      const rawResponse = await $.handleHttpRequest("get", dataFileUrl);
+      const urlPath = $.getRelativePathFromURL(dataFileUrl)
```
Comment (koladilip) [inline src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml:64]: this is good but it is a chance to make mistakes like people copy paste and URL might contain dynamic parts, I prefer hard coded endpoint paths

Comment (sanpj2292) [reply src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml:64]: I agree why you say hardcoded paths are needed. But in this case, I don't even know how does the URL path look like. Hence had to adopt this.

File: src/v0/util/index.js (hunk 1 above)
Comment (sanpj2292) [reply src/v0/util/index.js:2261]: sure

File: src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml (hunk 2 above)
Comment (koladilip) [reply src/cdk/v2/destinations/optimizely_fullstack/procWorkflow.yaml:64]: got it, it is coming from the config

File: src/v0/util/index.js (hunk 1 above)
Comment (sanpj2292) [reply src/v0/util/index.js:2269]: Done

**Review by koladilip — DISMISSED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Review by ItsSudip — DISMISSED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by Gauravudia — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**